src/index.ts

```diff
@@ -14,24 +16,58 @@ export class BaseError extends makeError.BaseError {
     super(message);
   }
 
-  [inspect.custom || "inspect"]() {
+  [customInspect]() {
     return fullStack(this);
   }
 }
 
 /**
  * Capture the full stack trace of any error instance.
  */
-export function fullStack(error: Error | BaseError) {
-  let err: Error | undefined = (error as BaseError).cause;
-  let fullStack = error.stack || error.message || "";
+export function fullStack(error: Error | BaseError): string {
+  const causeChain = getCauseChain(error);
+  const inspections = causeChain.map(e => inspectSingleCauseLink(e));
+  return inspections.join(SEPARATOR_TEXT);
+}
+
+/**
+ * Get an array of errors, with the first array entry being the given
+ * error itself, and all the causes following afterwards.
+ *
+ * @param error The error whose cause chain is to be extracted.
+ * @return An array containing the cause chain.
+ */
+function getCauseChain(error: Error | BaseError): Array<Error | BaseError> {
+  const links: Array<Error | BaseError> = [];
+  let currentError = error;
 
-  while (err) {
-    fullStack += SEPARATOR_TEXT;
-    fullStack += err.stack || err.message || "";
+  while (true) {
+    links.push(currentError);
 
-    err = (err as BaseError).cause;
+    if (currentError instanceof BaseError && currentError.cause !== undefined) {
+      currentError = currentError.cause;
+    } else {
+      break;
+    }
   }
 
-  return fullStack;
+  return links;
+}
+
+function inspectSingleCauseLink(error: Error | BaseError): string {
+  let properties = Object.getOwnPropertyDescriptors(error);
+
+  delete properties["cause"];
+  Reflect.set(properties, customInspect, {
+    writable: false,
+    configurable: false,
+    enumerable: false,
+    value: undefined
+  } as PropertyDescriptor);
+
+  // create a shallow copy, omitting "cause" and the custom inspect method
+  // so we only get this one error without its cause nicely formatted
+  // and with support for custom inspections for any data on the error
+  // (turning off customInspect is not necessary this way)
```

Comment:
Why did you _not_ want to turn off `customInspect` by the way?

Reply:
This was meant to be for the case when an error holds data with a custom inspect method.